db/views/reporting_appointment_scheduled_days_distributions_v01.sql

```diff
@@ -0,0 +1,26 @@
+WITH latest_appointments_per_patient_per_month AS (
+    SELECT DISTINCT ON (patient_id, month_date) *,
+        to_char(device_created_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE')), 'YYYY-MM-01')::date month_date
+    FROM appointments
+    WHERE extract('days' FROM (scheduled_date - device_created_at))::integer >= 0
+      AND device_created_at >= DATE_TRUNC('month', (now() AT TIME ZONE 'UTC') - INTERVAL '6 months')
+    ORDER BY patient_id, month_date desc
+),
+ scheduled_days_distribution AS (
+     SELECT month_date,
+            width_bucket(extract('days' FROM (scheduled_date - device_created_at))::integer, array[0, 15, 30, 60]) bucket,
+            COUNT(*) number_of_appointments,
+            creation_facility_id facility_id
+     FROM latest_appointments_per_patient_per_month
+     GROUP BY bucket, creation_facility_id, month_date
+)
+
+SELECT facility_id,
+       month_date,
+       (SUM(number_of_appointments) FILTER (WHERE bucket = 1))::integer AS appts_scheduled_0_to_14_days,
+       (SUM(number_of_appointments) FILTER (WHERE bucket = 2))::integer AS appts_scheduled_15_to_30_days,
+       (SUM(number_of_appointments) FILTER (WHERE bucket = 3))::integer AS appts_scheduled_31_to_60_days,
+       (SUM(number_of_appointments) FILTER (WHERE bucket = 4))::integer AS appts_scheduled_more_than_60_days,
+       SUM(number_of_appointments)::integer total_appts_scheduled
```

Comment:
> 1 record per appointment

The `distribution` in the name indicates that this is not individual appointment records. The name doesn't do a good job of explaining this has 1 record per facility per month though, yeah. I guess we want to say that it's a `distribution of the appointment's scheduled days in a facility`. Is this too long `reporting_facility_appointment_scheduled_days_distribution`?

Reply:
Haha that is quite a mouthful huh. My other ideas are
* `reporting_facility_follow_up_days_distribution` - "Follow-up days" is a concept we use right?
* `reporting_facility_appointment_scheduled_days` - We could drop the "distribution" since it's evident from the column names immediately

What do you think?

Reply:
We've avoided `follow_ups` since there's already a follow ups view. I think we can drop `distribution` for `facility` in the name, going with `reporting_facility_appointment_scheduled_days`.

Reply:
🚀

Reply:
Addressed in 8d3af49